src/assisted_test_infra/test_infra/controllers/node_controllers/vsphere_controller.py

```diff
@@ -28,6 +28,8 @@ def prepare_nodes(self):
         config = {**self._config.get_all(), **self._entity_config.get_all(), "cluster_name": self.cluster_name}
         # The ISO file isn't available now until preparing for installation
         del config["iso_download_path"]
+        folder_name = "assisted-test-infra-ci" if self._config.build_id is not None else "assisted-test-infra"
+        self._create_folder(folder_name)
```

Comment:
Why not to use the same mechanism as we do on `TerraformController` (`ClusterName.suffix`)?

Reply:
To make it easier to manage, we separated the test-infra clusters from the CI clusters. Each cluster was placed inside a folder with its build-id.
The mechanism is new, since our existing virtual machines ran on a bare-metal host. Here, the cluster is outside the bare metal host.

Reply:
It makes sense to group everything in one place since the CI (test-infra) machine is created within the same folder.